src/components/CommentForm.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useContext, useState } from 'react';
+import UserContext from '../contexts/UserContext';
+import { postNewComment } from '../api';
+import Loading from './Loading';
+
+const CommentForm = ({ article, setNewComment }) => {
+
+    const { currentUser } = useContext(UserContext);
+
+    const [newCommentInput, setNewCommentInput] = useState("");
+    const [newCommentInputError, setNewCommentInputError] = useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const handleNewCommentChange = (event) => {
+        setNewCommentInput(event.target.value);
+        if (!validateNewComment(event.target.value)) {
+            setNewCommentInputError("This field is required");
+        } else {
+            setNewCommentInputError(null);
+        }
+    }
+
+    const validateNewComment = (value) => {
+        return value.length > 0;
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        if (!validateNewComment(newCommentInput)) {
+            alert("You cannot submit an empty comment");
+            return;
+        }
+        setIsLoading(true);
+        postNewComment(article.article_id, currentUser, newCommentInput)
+        .then((comment) => {
+            alert("New Comment Added!");
+            // triggers a re-render in ArticleComments.jsx
```

Comment:
I like the concept! However from a UX perspective, the rerender of the component makes everything feel a little 'jerky'. If your API returns the posted comment, I would probably recommend just disabling the form until it's posted, getting the posted comment back from the API and setting the comments in state with the posted one on top